h/streamer/views.py

```diff
@@ -21,6 +21,19 @@ def websocket_view(request):
     if origin is not None:
         if origin != request.host_url and origin not in allowed:
             return httpexceptions.HTTPForbidden()
+
+    # Provide environment which the WebSocket handler can use...
+    request.environ.update({
+        'h.ws.authenticated_userid': request.authenticated_userid,
+        'h.ws.effective_principals': request.effective_principals,
+        'h.ws.registry': request.registry,
+        'h.ws.streamer_work_queue': streamer.WORK_QUEUE,
+    })
```

Comment:
I'm not sure if this is one of the points that you are a bit unhappy about, but I do like this approach. At this point we're handing the connection off to a non-pyramid part of our application, so we make sure to pass the necessary data on instead of _magically_ getting a request.